admin-dev/themes/new-theme/js/pages/product/edit/specific-prices-manager.ts

```diff
@@ -76,15 +80,67 @@ export default class SpecificPricesManager {
     );
 
     this.initSpecificPriceModals();
-
-    // Enable/disable the priority selectors depending on the priority type selected (global or custom)
+    this.initCatalogPriceRules();
+    // Enable/disabled the priority selectors depending on the priority type selected (global or custom)
     new FormFieldToggler({
       disablingInputSelector: PriorityMap.priorityTypeCheckboxesSelector,
       matchingValue: '0',
       targetSelector: PriorityMap.priorityListWrapper,
     });
   }
 
+  private initCatalogPriceRules() {
+    const priceRuleRenderer = new CatalogPriceRuleRenderer();
+    const catalogPriceRulePaginator = new DynamicPaginator(
+      CatalogPriceRulesMap.paginationContainer,
+      new PaginatedCatalogPriceRulesService(this.productId),
+      priceRuleRenderer,
+      1,
+    );
+
+    const showCatalogPriceRulesButton = document.querySelector<HTMLElement>(CatalogPriceRulesMap.showCatalogPriceRules);
+    const catalogPriceRulesContainer = document.querySelector<HTMLElement>(CatalogPriceRulesMap.blockContainer);
+    let listRendered = false;
+    let listShown = false;
+
+    if (showCatalogPriceRulesButton === null) {
+      console.log(`Error: ${CatalogPriceRulesMap.showCatalogPriceRules} element not found`);
+      return;
+    }
+    if (catalogPriceRulesContainer === null) {
+      console.log(`Error: ${CatalogPriceRulesMap.blockContainer} element not found`);
+      return;
+    }
+    const showIcon = showCatalogPriceRulesButton.dataset.showIcon ?? 'visibility';
+    const hideIcon = showCatalogPriceRulesButton.dataset.hideIcon ?? 'visibility_off';
+    const showLabel = showCatalogPriceRulesButton.dataset.showLabel ?? 'Show catalog price rules';
+    const hideLabel = showCatalogPriceRulesButton.dataset.hideLabel ?? 'Hide catalog price rules';
+
+    /** This should be the form container for the whole form element, so we can hide the whole block */
+    const formContainer = <HTMLElement>catalogPriceRulesContainer.parentNode;
+
+    if (formContainer === null) {
+      console.log(`Error: ${CatalogPriceRulesMap.blockContainer} parent element not found`);
+      return;
+    }
+
+    showCatalogPriceRulesButton.addEventListener('click', () => {
+      formContainer.classList.toggle('d-none');
+      if (!listShown) {
+        showCatalogPriceRulesButton.innerHTML = `<i class="material-icons"> ${hideIcon}</i> ${hideLabel}`;
+        listShown = true;
+      } else {
+        showCatalogPriceRulesButton.innerHTML = `<i class="material-icons"> ${showIcon}</i> ${showLabel}`;
```

Comment:
```suggestion
        showCatalogPriceRulesButton.innerHTML = `<i class="material-icons">visibility_off</i> ${showLabel}`;
```

As I suggested in the form type, I'm not sure this one needs to be configured via data attributes since it's likely not to change ever Unlike the label which needs to be translated via PHP

Reply:
Hmm wanted to maintain ability to change it from PHP, but you are right icon name probably never gonna change, will add.

Reply:
Added this